src/Workspaces/CoreTest/Renamer/RenamerTests.cs

```diff
@@ -0,0 +1,134 @@
+
+                documentIdToDocumentInfoMap.Add((documentId, endDocuments[i]));
+            }
+
+            foreach (var (documentId, endDocument) in documentIdToDocumentInfoMap)
+            {
+                var document = solution.GetDocument(documentId);
+                var documentInfo = DocumentInfo.Create(document)
+                    .WithName(endDocument.DocumentName)
+                    .WithFolders(endDocument.DocumentFolders)
+                    .WithFilePath(endDocument.DocumentFilePath);
+
+                var result = await Rename.Renamer.RenameDocumentAsync(document, documentInfo, workspace.Options);
+                solution = result.Solution;
+
+                var updatedDocument = solution.GetDocument(documentId);
+
+                Assert.Equal(endDocument.DocumentName, updatedDocument.Name);
+                Assert.Equal(endDocument.DocumentFilePath, updatedDocument.FilePath);
+                AssertEx.SetEqual(endDocument.DocumentFolders, updatedDocument.Folders);
+                AssertEx.EqualOrDiff(endDocument.Text, (await updatedDocument.GetTextAsync()).ToString());
+                Assert.Equal(endDocument.ExpectRename, result.RenameSymbolSucceeded);
+            }
+        }
+
+        private static string[] GetDocumentFolders(string filePath)
+        {
+            var splitPath = filePath.Split('\\');
+            if (splitPath.Length == 1)
+            {
+                return splitPath;
+            }
+
+            return splitPath.Take(splitPath.Length - 1).ToArray();
+        }
+
+        private static Task TestRenameDocument(string startText, string expectedText, string languageName, string newDocumentName = null, string newDocumentPath = null, string documentName = null, string documentPath = null, bool expectRename = true)
+        {
+            var defaultDocumentName = documentName ?? DefaultDocumentName;
+            var defaultDocumentPath = documentPath ?? DefaultDocumentPath;
+
+            var startDocuments = new[]
+            {
+                new DocumentWithInfo()
+                {
+                    Text = startText,
+                    DocumentName = defaultDocumentName,
+                    DocumentFilePath = defaultDocumentPath
+                }
+            };
+
+            var endDocuments = new[]
+            {
+                new DocumentWithInfo()
+                {
+                    Text = expectedText,
+                    DocumentName = newDocumentName ?? defaultDocumentName,
+                    DocumentFilePath = newDocumentPath ?? defaultDocumentPath,
+                    ExpectRename = expectRename
+                }
+            };
+
+            return TestRenameDocument(startDocuments, endDocuments, languageName);
+        }
+
+        private static Task TestRenameDocumentCSharp(DocumentWithInfo[] startDocuments, DocumentWithInfo[] expectedDocuments)
+        => TestRenameDocument(startDocuments, expectedDocuments, LanguageNames.CSharp);
```

Comment:
Rather than the C#/VB overloads, consider just having an abstract member that returns the language name, and the helpers just use that. Each language overload just gets to say it once then.